File: javascript/bob/bob.js
export const hey = (message) => {
  let isUpperCase = /[a-zA-Z]/.test(message) && message === message.toUpperCase();
  let isQuestion = message.trim().endsWith('?');
  if (isUpperCase && isQuestion) {
    return 'Calm down, I know what I\'m doing!';
  } else if (isUpperCase) {
    return 'Whoa, chill out!';
  } else if (isQuestion) {
    return 'Sure.';
  } else if (/\S/.test(message)) {
    return 'Whatever.';
  } else {
    return 'Fine. Be that way!';
  };
};

File: python/bob/bob.py
def response(hey_bob):
    hey_bob = hey_bob.rstrip()
    if not hey_bob:
        return 'Fine. Be that way!'
    is_upper = hey_bob.isupper()
    is_question = hey_bob.endswith('?')
    if is_upper and is_question:
        return 'Calm down, I know what I\'m doing!'
    if is_upper:
        return 'Whoa, chill out!'
    if is_question:
        return 'Sure.'
    return 'Whatever.'
